test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperOutputTest.cs

```diff
@@ -33,329 +32,42 @@ public void TagName_CanSetToNull()
-                },
-                htmlEncoder: new NullHtmlEncoder());
-
-            tagHelperOutput.SelfClosing = true;
-
-            // Act
-            var output = tagHelperOutput.GenerateStartTag();
-
-            // Assert
-            Assert.Equal("<p class=\"btn\" something=\"   spaced    \" />", output);
-        }
-
-        [Fact]
-        public void GenerateStartTag_ReturnsSelfClosingStartTag_NoAttributes()
-        {
-            // Arrange
-            var tagHelperOutput = new TagHelperOutput("p");
-
-            tagHelperOutput.SelfClosing = true;
-
-            // Act
-            var output = tagHelperOutput.GenerateStartTag();
-
-            // Assert
-            Assert.Equal("<p />", output);
-        }
-
-        [Fact]
-        public void GenerateStartTag_UsesProvidedHtmlEncoder()
-        {
-            // Arrange
-            var tagHelperOutput = new TagHelperOutput("p",
-                attributes: new Dictionary<string, string>
-                {
-                    { "hello", "world" },
-                },
-                htmlEncoder: new PseudoHtmlEncoder());
-
-            tagHelperOutput.SelfClosing = true;
-
-            // Act
-            var output = tagHelperOutput.GenerateStartTag();
-
-            // Assert
-            Assert.Equal("<p hello=\"HtmlEncode[[world]]\" />", output);
-        }
-
-        [Fact]
-        public void GenerateStartTag_ReturnsNothingIfWhitespaceTagName()
-        {
-            // Arrange
-            var tagHelperOutput = new TagHelperOutput("  ",
-                attributes: new Dictionary<string, string>
-                {
-                    { "class", "btn" },
-                    { "something", "   spaced    " }
-                },
-                htmlEncoder: new NullHtmlEncoder())
-            {
-                SelfClosing = true
-            };
-
-            // Act
-            var output = tagHelperOutput.GenerateStartTag();
-
-            // Assert
-            Assert.Empty(output);
-        }
-
-
-        [Fact]
-        public void GenerateEndTag_ReturnsNothingIfWhitespaceTagName()
-        {
-            // Arrange
-            var tagHelperOutput = new TagHelperOutput(" ");
-            tagHelperOutput.Content.Append("Hello World");
-
-            // Act
-            var output = tagHelperOutput.GenerateEndTag();
-
-            // Assert
-            Assert.Empty(output);
-        }
-
-        [Fact]
-        public void GeneratePreContent_ReturnsPreContent()
-        {
-            // Arrange
-            var tagHelperOutput = new TagHelperOutput("p");
-            tagHelperOutput.PreContent.Append("Hello World");
-
-            // Act
-            var output = tagHelperOutput.GeneratePreContent();
-
-            // Assert
-            var result = Assert.IsType<DefaultTagHelperContent>(output);
+            // Act & Assert
+            var result = Assert.IsType<DefaultTagHelperContent>(tagHelperOutput.PreContent);
```

Comment:
this assertion confirms a low-level detail (the implementation type).  should instead `Assert.NotNull(tagHelperOutput.PreContent)`